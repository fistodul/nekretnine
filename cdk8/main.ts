import { Construct } from 'constructs';
import { App, Chart, ChartProps } from 'cdk8s';

import * as fs from 'fs';
import * as dotenv from 'dotenv';
import * as path from 'path';

import {
  KubeDeployment,
  KubeService,
  KubeConfigMap,
  IntOrString
} from './imports/k8s';

// Load the .env file from ../.env
dotenv.config({ path: path.join(__dirname, '..', '.env') });

export class NekretnineChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);

    // Create a ConfigMap to hold the SQL initialization scripts (without PersistentVolume)
    const sqlConfigMap = new KubeConfigMap(this, `${id}-sql-configmap`, {
      data: {
        'a.sql': fs.readFileSync(path.join(__dirname, '..', 'compose', 'nekretnine.sql')).toString(),
        'b.sql': fs.readFileSync(path.join(__dirname, '..', 'compose', 'data.sql')).toString()
      }
    });

    // Define the database (MariaDB) deployment
    new KubeDeployment(this, 'db-deployment', {
      spec: {
        replicas: 1,
        selector: {
          matchLabels: { app: `${id}-db` }
        },
        template: {
          metadata: { labels: { app: `${id}-db` } },
          spec: {
            containers: [
              {
                name: 'db',
                image: 'mariadb',
                ports: [{ containerPort: Number(process.env.DB_PORT) || 3306 }],
                env: [
                  { name: 'MARIADB_USER', value: process.env.DB_USER || 'testuser123' },
                  { name: 'MARIADB_PASSWORD', value: process.env.DB_PASS || 'testpass123' },
                  { name: 'MARIADB_DATABASE', value: process.env.DB_DB || 'nekretnine' },
                  { name: 'MARIADB_RANDOM_ROOT_PASSWORD', value: '1' }
                ],
                livenessProbe: {
                  exec: {
                    command: [
                      'healthcheck.sh',
                      '--su-mysql',
                      '--connect',
                      '--innodb_initialized'
                    ]
                  },
                  initialDelaySeconds: 15,
                  periodSeconds: 15,
                  timeoutSeconds: 30,
                  failureThreshold: 3
                },
                volumeMounts: [
                  {
                    name: 'init-sql',
                    mountPath: '/docker-entrypoint-initdb.d'
                  }
                ]
              }
            ],
            volumes: [
              {
                name: 'init-sql',
                configMap: {
                  // Use the name from the ConfigMap resource
                  name: sqlConfigMap.name
                }
              }
            ]
          }
        }
      }
    });

    // Service for the database – accessible only inside the cluster
    new KubeService(this, 'db-service', {
      spec: {
        type: 'ClusterIP',
        ports: [
          { port: 3306, targetPort: IntOrString.fromNumber(3306) }
        ],
        selector: { app: `${id}-db` }
      }
    });

    // Define the web application deployment
    new KubeDeployment(this, 'web-deployment', {
      spec: {
        replicas: 1,
        selector: {
          matchLabels: { app: `${id}-web` }
        },
        template: {
          metadata: { labels: { app: `${id}-web` } },
          spec: {
            containers: [
              {
                name: 'web',
                image: 'filipmania/nekretnine',
                ports: [{ containerPort: Number(process.env.WEB_PORT) || 8000 }],
                env: [
                  { name: 'NODE_ENV', value: process.env.NODE_ENV || 'production' },
                  { name: 'DB_HOST', value: process.env.DB_HOST || 'db' },
                  { name: 'DB_USER', value: process.env.DB_USER || 'testuser123' },
                  { name: 'DB_PASS', value: process.env.DB_PASS || 'testpass123' },
                  { name: 'DB_DB', value: process.env.DB_DB || 'nekretnine' },
                  { name: 'DB_PORT', value: process.env.DB_PORT || '3306' },
                  { name: 'COOKIE_SECRET', value: process.env.COOKIE_SECRET || 'testcookie123' },
                  { name: 'WEB_PASS', value: process.env.WEB_PASS || 'testpass123' },
                  { name: 'WEB_PORT', value: process.env.WEB_PORT || '8000' }
                ]
              }
            ]
          }
        }
      }
    });

    // Service for the web deployment – exposing it externally via a LoadBalancer
    new KubeService(this, 'web-service', {
      spec: {
        type: 'LoadBalancer',
        ports: [
          { port: 8000, targetPort: IntOrString.fromNumber(8000) }
        ],
        selector: { app: `${id}-web` }
      }
    });

  }
}

// Instantiate the app and synthesize the Kubernetes manifests
const app = new App();
new NekretnineChart(app, 'nekretnine');
app.synth();
