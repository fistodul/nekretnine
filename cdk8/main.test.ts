import {NekretnineChart} from './main';
import {Testing} from 'cdk8s';

describe('Placeholder', () => {
  test('Empty', () => {
    const app = Testing.app();
    const chart = new NekretnineChart(app, 'test-chart');
    const results = Testing.synth(chart)
    expect(results).toMatchSnapshot();
  });
});
