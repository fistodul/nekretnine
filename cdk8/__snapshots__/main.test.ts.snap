// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Empty 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "data": Object {
      "a.sql": "-- MariaDB dump 10.18  Distrib 10.5.8-MariaDB, for Linux (x86_64)
--
-- Host: localhost    Database: nekretnine
-- ------------------------------------------------------
-- Server version	10.5.8-MariaDB

/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!40101 SET @OLD_CHARACTER_SET_RESULTS=@@CHARACTER_SET_RESULTS */;
/*!40101 SET @OLD_COLLATION_CONNECTION=@@COLLATION_CONNECTION */;
/*!40101 SET NAMES utf8mb4 */;
/*!40103 SET @OLD_TIME_ZONE=@@TIME_ZONE */;
/*!40103 SET TIME_ZONE='+00:00' */;
/*!40014 SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0 */;
/*!40014 SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 */;
/*!40101 SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='NO_AUTO_VALUE_ON_ZERO' */;
/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;

--
-- Table structure for table \`lokacije\`
--

DROP TABLE IF EXISTS \`lokacije\`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE \`lokacije\` (
  \`id\` smallint(5) unsigned NOT NULL AUTO_INCREMENT,
  \`ime\` varchar(20) COLLATE utf8mb4_unicode_ci NOT NULL,
  PRIMARY KEY (\`id\`)
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Table structure for table \`nekretnine\`
--

DROP TABLE IF EXISTS \`nekretnine\`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE \`nekretnine\` (
  \`id\` smallint(5) unsigned NOT NULL AUTO_INCREMENT,
  \`v_id\` tinyint(3) unsigned NOT NULL,
  \`ime\` varchar(35) COLLATE utf8mb4_unicode_ci NOT NULL,
  PRIMARY KEY (\`id\`),
  KEY \`FK_nekretnine_vrste\` (\`v_id\`) USING BTREE,
  CONSTRAINT \`FK_nekretnine_vrste\` FOREIGN KEY (\`v_id\`) REFERENCES \`vrste\` (\`id\`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Table structure for table \`poruke\`
--

DROP TABLE IF EXISTS \`poruke\`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE \`poruke\` (
  \`id\` mediumint(8) unsigned NOT NULL AUTO_INCREMENT,
  \`p_id\` mediumint(8) unsigned DEFAULT NULL,
  \`poruka\` text COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  PRIMARY KEY (\`id\`),
  KEY \`poruke_FK\` (\`p_id\`),
  CONSTRAINT \`poruke_FK\` FOREIGN KEY (\`p_id\`) REFERENCES \`prostori\` (\`id\`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Table structure for table \`prostori\`
--

DROP TABLE IF EXISTS \`prostori\`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE \`prostori\` (
  \`id\` mediumint(8) unsigned NOT NULL AUTO_INCREMENT,
  \`n_id\` smallint(6) unsigned NOT NULL,
  \`izdavanje\` tinyint(1) NOT NULL,
  \`broj_soba\` float NOT NULL,
  \`cena\` mediumint(8) unsigned DEFAULT NULL,
  \`stara_cena\` mediumint(8) unsigned DEFAULT NULL,
  \`kvadrati\` smallint(6) unsigned DEFAULT NULL,
  \`spratnost\` tinyint(4) unsigned DEFAULT NULL,
  \`karakteristike\` varchar(300) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  \`opis\` text COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  \`lokacija\` varchar(37) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  \`currency\` varchar(3) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT 'EUR',
  \`top\` tinyint(1) DEFAULT NULL,
  PRIMARY KEY (\`id\`),
  KEY \`FK_prostori_nekretnine\` (\`n_id\`),
  KEY \`prostori_broj_soba_IDX\` (\`broj_soba\`) USING BTREE,
  KEY \`prodaja\` (\`izdavanje\`) USING BTREE,
  CONSTRAINT \`FK_prostori_nekretnine\` FOREIGN KEY (\`n_id\`) REFERENCES \`nekretnine\` (\`id\`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Table structure for table \`prostorulice\`
--

DROP TABLE IF EXISTS \`prostorulice\`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE \`prostorulice\` (
  \`id\` mediumint(8) unsigned NOT NULL AUTO_INCREMENT,
  \`p_id\` mediumint(8) unsigned DEFAULT NULL,
  \`u_id\` mediumint(8) unsigned DEFAULT NULL,
  PRIMARY KEY (\`id\`),
  KEY \`FK_prostor-ulice_prostori\` (\`p_id\`),
  KEY \`FK_prostor-ulice_ulice\` (\`u_id\`),
  CONSTRAINT \`FK_prostor-ulice_prostori\` FOREIGN KEY (\`p_id\`) REFERENCES \`prostori\` (\`id\`) ON DELETE CASCADE ON UPDATE CASCADE,
  CONSTRAINT \`FK_prostor-ulice_ulice\` FOREIGN KEY (\`u_id\`) REFERENCES \`ulice\` (\`id\`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Table structure for table \`reklame\`
--

DROP TABLE IF EXISTS \`reklame\`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE \`reklame\` (
  \`id\` smallint(5) unsigned NOT NULL AUTO_INCREMENT,
  \`poruka\` text COLLATE utf8mb4_unicode_ci NOT NULL,
  \`slika\` varchar(80) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  \`link\` varchar(80) COLLATE utf8mb4_unicode_ci DEFAULT NULL,
  PRIMARY KEY (\`id\`)
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Table structure for table \`slike\`
--

DROP TABLE IF EXISTS \`slike\`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE \`slike\` (
  \`id\` mediumint(8) unsigned NOT NULL AUTO_INCREMENT,
  \`p_id\` mediumint(8) unsigned NOT NULL,
  \`slika\` varchar(80) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
  PRIMARY KEY (\`id\`),
  KEY \`FK_slike_prostori\` (\`p_id\`),
  CONSTRAINT \`slike_FK\` FOREIGN KEY (\`id\`) REFERENCES \`prostori\` (\`id\`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Temporary table structure for view \`top_ponude\`
--

DROP TABLE IF EXISTS \`top_ponude\`;
/*!50001 DROP VIEW IF EXISTS \`top_ponude\`*/;
SET @saved_cs_client     = @@character_set_client;
SET character_set_client = utf8;
/*!50001 CREATE TABLE \`top_ponude\` (
  \`id\` tinyint NOT NULL,
  \`cena\` tinyint NOT NULL,
  \`stara_cena\` tinyint NOT NULL,
  \`kvadrati\` tinyint NOT NULL,
  \`broj_soba\` tinyint NOT NULL,
  \`spratnost\` tinyint NOT NULL,
  \`lokacija\` tinyint NOT NULL,
  \`currency\` tinyint NOT NULL
) ENGINE=MyISAM */;
SET character_set_client = @saved_cs_client;

--
-- Table structure for table \`ulice\`
--

DROP TABLE IF EXISTS \`ulice\`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE \`ulice\` (
  \`id\` mediumint(8) unsigned NOT NULL AUTO_INCREMENT,
  \`l_id\` smallint(5) unsigned NOT NULL,
  \`ime\` varchar(35) COLLATE utf8mb4_unicode_ci NOT NULL,
  PRIMARY KEY (\`id\`),
  KEY \`FK_ulice_lokacije\` (\`l_id\`),
  CONSTRAINT \`FK_ulice_lokacije\` FOREIGN KEY (\`l_id\`) REFERENCES \`lokacije\` (\`id\`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Table structure for table \`vrste\`
--

DROP TABLE IF EXISTS \`vrste\`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!40101 SET character_set_client = utf8 */;
CREATE TABLE \`vrste\` (
  \`id\` tinyint(3) unsigned NOT NULL AUTO_INCREMENT,
  \`ime\` varchar(20) COLLATE utf8mb4_unicode_ci NOT NULL,
  PRIMARY KEY (\`id\`)
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
/*!40101 SET character_set_client = @saved_cs_client */;

--
-- Final view structure for view \`top_ponude\`
--

/*!50001 DROP TABLE IF EXISTS \`top_ponude\`*/;
/*!50001 DROP VIEW IF EXISTS \`top_ponude\`*/;
/*!50001 SET @saved_cs_client          = @@character_set_client */;
/*!50001 SET @saved_cs_results         = @@character_set_results */;
/*!50001 SET @saved_col_connection     = @@collation_connection */;
/*!50001 SET character_set_client      = utf8mb4 */;
/*!50001 SET character_set_results     = utf8mb4 */;
/*!50001 SET collation_connection      = utf8mb4_general_ci */;
/*!50001 CREATE ALGORITHM=UNDEFINED */
/*!50013  SQL SECURITY DEFINER */
/*!50001 VIEW \`top_ponude\` AS select \`prostori\`.\`id\` AS \`id\`,\`prostori\`.\`cena\` AS \`cena\`,\`prostori\`.\`stara_cena\` AS \`stara_cena\`,\`prostori\`.\`kvadrati\` AS \`kvadrati\`,\`prostori\`.\`broj_soba\` AS \`broj_soba\`,\`prostori\`.\`spratnost\` AS \`spratnost\`,\`prostori\`.\`lokacija\` AS \`lokacija\`,\`prostori\`.\`currency\` AS \`currency\` from \`prostori\` where \`prostori\`.\`top\` = 1 */;
/*!50001 SET character_set_client      = @saved_cs_client */;
/*!50001 SET character_set_results     = @saved_cs_results */;
/*!50001 SET collation_connection      = @saved_col_connection */;
/*!40103 SET TIME_ZONE=@OLD_TIME_ZONE */;

/*!40101 SET SQL_MODE=@OLD_SQL_MODE */;
/*!40014 SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS */;
/*!40014 SET UNIQUE_CHECKS=@OLD_UNIQUE_CHECKS */;
/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;
/*!40101 SET CHARACTER_SET_RESULTS=@OLD_CHARACTER_SET_RESULTS */;
/*!40101 SET COLLATION_CONNECTION=@OLD_COLLATION_CONNECTION */;
/*!40111 SET SQL_NOTES=@OLD_SQL_NOTES */;

-- Dump completed on 2020-11-21 18:37:35
",
      "b.sql": "REPLACE INTO vrste (id, ime) VALUES
(1, 'Stambeni prostor'),
(2, 'Poslovni prostor');

REPLACE INTO nekretnine (id, v_id, ime) VALUES
(1, 1, 'Stan'),
(2, 1, 'Soba'),
(3, 1, 'Kuća'),
(4, 1, 'Garaža'),
(5, 1, 'Zemljište'),
(6, 1, 'Ostalo...'),
(7, 2, 'Lokal'),
(8, 2, 'Kancelarijski / Poslovni prostor'),
(9, 2, 'Poslovna zgrada'),
(10, 2, 'Hala'),
(11, 2, 'Magacin'),
(12, 2, 'Ugostiteljski objekat'),
(13, 2, 'Kiosk'),
(14, 2, 'Stovarište'),
(15, 2, 'Ostalo...');

REPLACE INTO lokacije (id, ime) VALUES
(1, 'Crveni Krst'),
(2, 'Medijana'),
(3, 'Niška Banja'),
(4, 'Palilula'),
(5, 'Pantelej'),
(6, 'Ostali gradovi');

REPLACE INTO ulice (id, l_id, ime) VALUES
(1, 1, 'Crveni Krst'),
(2, 1, 'Aerodrom'),
(3, 1, 'Autobuska stanica'),
(4, 1, 'Beograd Mala'),
(5, 1, 'Berčinac'),
(6, 1, 'Branko Bjegović'),
(7, 1, 'Bulevar Nikole Tesle'),
(8, 1, 'Čamurlija'),
(9, 1, 'Donja Toponica'),
(10, 1, 'Donja Trnava'),
(11, 1, 'Donji Komren'),
(12, 1, 'Gornja Toponica'),
(13, 1, 'Gornja Trnava'),
(14, 1, 'Hum'),
(15, 1, 'Humska'),
(16, 1, 'Jagodin Mala'),
(17, 1, 'Kravlje'),
(18, 1, 'Leskovik'),
(19, 1, 'Medoševac'),
(20, 1, 'Mezgraja'),
(21, 1, 'Miljkovac'),
(22, 1, 'Paligrace'),
(23, 1, 'Paljina'),
(24, 1, 'Podvinik'),
(25, 1, 'Popovac'),
(26, 1, 'Ratko Jović'),
(27, 1, 'Rujnik'),
(28, 1, 'Sečanica'),
(29, 1, 'Supovac'),
(30, 1, 'Šljaka'),
(31, 1, 'Tehnički fakulteti'),
(32, 1, 'Trupale'),
(33, 1, 'Tvrđava'),
(34, 1, 'Vele Polje'),
(35, 1, 'Velika pijaca'),
(36, 1, 'Vrtište'),
(37, 2, 'Medijana'),
(38, 2, 'Božidara Adžije'),
(39, 2, 'Brzi Brod'),
(40, 2, 'Bulevar dr Zorana Đinđića'),
(41, 2, 'Bulevar Nemanjića'),
(42, 2, 'Bulevar zona 1'),
(43, 2, 'Bulevar zona 2'),
(44, 2, 'Bulevar zona 3'),
(45, 2, 'Centar'),
(46, 2, 'Crveni pevac'),
(47, 2, 'Čair'),
(48, 2, 'Ćele kula'),
(49, 2, 'Dom zdravlja'),
(50, 2, 'Dušanov bazar'),
(51, 2, 'Duvanište'),
(52, 2, 'Filozofski fakultet'),
(53, 2, 'Gradska bolnica'),
(54, 2, 'Kičevo'),
(55, 2, 'Krive livade'),
(56, 2, 'Krivi Vir'),
(57, 2, 'Marger'),
(58, 2, 'Medicinski fakultet'),
(59, 2, 'Merkator'),
(60, 2, 'Obilićev venac'),
(61, 2, 'Pravni fakultet'),
(62, 2, 'Sinđelićev trg'),
(63, 2, 'Šojka'),
(64, 2, 'Trg Kralja Aleksandra'),
(65, 2, 'Trg Učitelja Tase'),
(66, 2, 'Trošarina'),
(67, 2, 'Vatrogasni dom'),
(68, 2, 'Vojna bolnica'),
(69, 3, 'Niška Banja'),
(70, 3, 'Bancarevo'),
(71, 3, 'Čuklenik'),
(72, 3, 'Donja Studena'),
(73, 3, 'Gornja Studena'),
(74, 3, 'Jelašnica'),
(75, 3, 'Koritnjak'),
(76, 3, 'Kunovica'),
(77, 3, 'Lazarevo Selo'),
(78, 3, 'Manastir'),
(79, 3, 'Nikola Tesla'),
(80, 3, 'Ostrovica'),
(81, 3, 'Prosek'),
(82, 3, 'Prva Kutina'),
(83, 3, 'Radikina Bara'),
(84, 3, 'Rautovo'),
(85, 3, 'Ravni Do'),
(86, 3, 'Sićevo'),
(87, 3, 'Ženeva (Devizno naselje)'),
(88, 4, 'Palilula'),
(89, 4, '9. Maj'),
(90, 4, 'Apelovac'),
(91, 4, 'Berbatovo'),
(92, 4, 'Bogoslovija'),
(93, 4, 'Bubanj / Bubanjska dolina'),
(94, 4, 'Crni put'),
(95, 4, 'Crvena Zvezda'),
(96, 4, 'Čokot'),
(97, 4, 'Delijski Vis'),
(98, 4, 'Donje Vlase'),
(99, 4, 'Durmitorska'),
(100, 4, 'Elektronska Industirja'),
(101, 4, 'Fakultet zaštite na radu'),
(102, 4, 'Gabrovac'),
(103, 4, 'Gabrovačka reka'),
(104, 4, 'Gabrovački put'),
(105, 4, 'Kalač brdo'),
(106, 4, 'Kovanluk'),
(107, 4, 'KP Dom'),
(108, 4, 'Krušce'),
(109, 4, 'Lalinac'),
(110, 4, 'Ledena Stena'),
(111, 4, 'Mašinska Industirja'),
(112, 4, 'Međurovo'),
(113, 4, 'Milka Protić'),
(114, 4, 'Mokranjčeva'),
(115, 4, 'Monopol'),
(116, 4, 'Mramor'),
(117, 4, 'Mramorski Potok'),
(118, 4, 'Nova železnička kolonija'),
(119, 4, 'Njegoševa'),
(120, 4, 'Pasi Poljana'),
(121, 4, 'PMF'),
(122, 4, 'Put za Medoševac'),
(123, 4, 'Rasadnik'),
(124, 4, 'Staro groblje'),
(125, 4, 'Stočni trg'),
(126, 4, 'Suvi Do'),
(127, 4, 'Svetinikolska crkva'),
(128, 4, 'Tutunović podrum'),
(129, 4, 'Ured'),
(130, 4, 'Veterinarska stanica'),
(131, 4, 'Vukmanovo'),
(132, 4, 'Železnička stanica'),
(133, 5, 'Pantelej'),
(134, 5, 'Beverli Hils'),
(135, 5, 'Brenica'),
(136, 5, 'Cerje'),
(137, 5, 'Čalije'),
(138, 5, 'Donja Vrežina'),
(139, 5, 'Donji Matejevac'),
(140, 5, 'Durlan'),
(141, 5, 'Gornja Vrežina'),
(142, 5, 'Gornji Matejevac'),
(143, 5, 'Jagodin Mala'),
(144, 5, 'Jasenovik'),
(145, 5, 'Kamenica'),
(146, 5, 'Knez selo'),
(147, 5, 'Knjaževačka'),
(148, 5, 'Malča'),
(149, 5, 'Oreovac'),
(150, 5, 'Pantelej'),
(151, 5, 'Pasjača'),
(152, 5, 'Somborska'),
(153, 5, 'Vinik'),
(154, 5, 'Voćarska'),
(155, 5, 'Vrelo'),
(156, 6, 'Ostali gradovi');
",
    },
    "kind": "ConfigMap",
    "metadata": Object {
      "name": "test-chart-test-chart-sql-configmap-c8aecf2a",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "test-chart-db-deployment-c86d93d3",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "test-chart-db",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "test-chart-db",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "MARIADB_USER",
                  "value": "testuser123",
                },
                Object {
                  "name": "MARIADB_PASSWORD",
                  "value": "testpass123",
                },
                Object {
                  "name": "MARIADB_DATABASE",
                  "value": "nekretnine",
                },
                Object {
                  "name": "MARIADB_RANDOM_ROOT_PASSWORD",
                  "value": "1",
                },
              ],
              "image": "mariadb",
              "livenessProbe": Object {
                "exec": Object {
                  "command": Array [
                    "healthcheck.sh",
                    "--su-mysql",
                    "--connect",
                    "--innodb_initialized",
                  ],
                },
                "failureThreshold": 3,
                "initialDelaySeconds": 15,
                "periodSeconds": 15,
                "timeoutSeconds": 30,
              },
              "name": "db",
              "ports": Array [
                Object {
                  "containerPort": 3306,
                },
              ],
              "volumeMounts": Array [
                Object {
                  "mountPath": "/docker-entrypoint-initdb.d",
                  "name": "init-sql",
                },
              ],
            },
          ],
          "volumes": Array [
            Object {
              "configMap": Object {
                "name": "test-chart-test-chart-sql-configmap-c8aecf2a",
              },
              "name": "init-sql",
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "test-chart-db-service-c8056ec5",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 3306,
          "targetPort": 3306,
        },
      ],
      "selector": Object {
        "app": "test-chart-db",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "test-chart-web-deployment-c8205f09",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "app": "test-chart-web",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "app": "test-chart-web",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "NODE_ENV",
                  "value": "test",
                },
                Object {
                  "name": "DB_HOST",
                  "value": "db",
                },
                Object {
                  "name": "DB_USER",
                  "value": "testuser123",
                },
                Object {
                  "name": "DB_PASS",
                  "value": "testpass123",
                },
                Object {
                  "name": "DB_DB",
                  "value": "nekretnine",
                },
                Object {
                  "name": "DB_PORT",
                  "value": "3306",
                },
                Object {
                  "name": "COOKIE_SECRET",
                  "value": "testcookie123",
                },
                Object {
                  "name": "WEB_PASS",
                  "value": "testpass123",
                },
                Object {
                  "name": "WEB_PORT",
                  "value": "8000",
                },
              ],
              "image": "filipmania/nekretnine",
              "name": "web",
              "ports": Array [
                Object {
                  "containerPort": 8000,
                },
              ],
            },
          ],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "test-chart-web-service-c85c9bdc",
    },
    "spec": Object {
      "ports": Array [
        Object {
          "port": 8000,
          "targetPort": 8000,
        },
      ],
      "selector": Object {
        "app": "test-chart-web",
      },
      "type": "LoadBalancer",
    },
  },
]
`;
